File: lalburst/swig/octave/check_lalburstswig.m
## Check SWIG Octave module wrapping lalburst
## Author: Karl Wette, 2011, 2012

## check module load
lalburst;
assert(exist("lalburst", "var"));
assert(exist("lalburstcvar", "var"));
lal;
assert(exist("lal", "var"));
assert(exist("lalcvar", "var"));
lalcvar.lalDebugLevel = 1;
disp("passed module load");

# check object parent tracking
a = lalburst.new_lalburstswig_test_parent_map_struct();
for i = 1:7
  b = a.s;
  c = lalburstcvar.lalburstswig_test_parent_map.s;
  lalburstcvar.lalburstswig_test_parent_map.s = lalcvar.lalswig_test_struct_const;
endfor
clear a b c;
CheckMemoryLeaks();
disp("passed object parent tracking");

## passed all tests!
disp("PASSED all tests");

File: lalburst/swig/python/check_lalburstswig.py
# Check SWIG Python module wrapping lalburst
# Author: Karl Wette, 2011, 2012

# check module load
print("checking module load ...")
import lal
import lalburst
from lalburst import cvar as lalburstcvar
from lal import cvar as lalcvar
print("PASSED module load")

# check object parent tracking
print("checking object parent tracking ...")
a = lalburst.swig_lalburst_test_parent_map_struct()
for i in range(0, 7):
    b = a.s
    c = lalburstcvar.swig_lalburst_test_parent_map.s
    lalburstcvar.swig_lalburst_test_parent_map.s = lalcvar.swig_lal_test_struct_const
del a, b, c
lal.CheckMemoryLeaks()
print("PASSED object parent tracking")

# passed all tests!
print("PASSED all tests")
